process.c

```diff
@@ -1051,6 +1055,10 @@ do_waitpid(rb_pid_t pid, int *st, int flags)
     return waitpid(pid, st, flags);
 #elif defined HAVE_WAIT4
     return wait4(pid, st, flags, NULL);
+#elif defined __wasi__
+    // WASI doesn't support spawning a process
+    errno = ENOTSUP;
+    return -1;
```

Comment:
What about defining `waitpid()` in wasm/missing.c?

Reply:
Makes sense to me 👍